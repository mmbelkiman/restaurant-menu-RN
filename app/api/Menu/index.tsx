export interface IApiMenuItem {
  name: string;
  description: string;
  price: number;
  url: string;
}

export interface ILocation {
  lat: string;
  log: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  postalCode: string;
  country: string;
}

export interface IApiMenu {
  id: string;
  name: string;
  description: string;
  phone: string;
  location: ILocation;
  operationDays: [];
  privateParking: boolean;
  currency: string;
  menus: Array<IApiMenus>;
}

export interface IApiMenus {
  name: string;
  items: Array<IApiMenuItem>;
}

export const Menu = {
  get: (): Promise<IApiMenu> => {
    let requestOptions = {
      method: 'GET',
      redirect: 'follow',
    };

    return new Promise((resolve, reject) => {
      fetch('https://mcdonalds.trio.dev/menu', requestOptions)
        .then(response => response.json())
        .then((result: IApiMenu) => resolve(result))
        .catch(() => reject(undefined));
    });
  },
};
