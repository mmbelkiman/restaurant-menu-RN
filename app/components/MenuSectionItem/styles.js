import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  carouselContainer: {
    marginTop: 30,
  },
  carouselItem: {
    textAlign: 'center',
    borderBottomWidth: 0,
    alignSelf: 'center',
    paddingBottom: 10,
    fontFamily: 'Montserrat-Regular',
  },
  carouselItemSelected: {
    textAlign: 'center',
    borderBottomWidth: 2,
    alignSelf: 'center',
    paddingBottom: 10,
    fontFamily: 'Montserrat-Regular',
  },
});
