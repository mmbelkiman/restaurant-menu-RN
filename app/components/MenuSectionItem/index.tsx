import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {IApiMenus} from '../../api/Menu';
import styles from './styles';

interface ILineSeparatorProps {
  index: number;
  item: IApiMenus;
  pressed: boolean;
  onPress(): void;
}

const MenuSectionItem: React.FC<ILineSeparatorProps> = ({
  index,
  item,
  pressed,
  onPress,
}) => {
  return (
    <TouchableOpacity
      testID={`menu-section-item-touchable-${index}`}
      style={styles.carouselContainer}
      onPress={() => onPress()}>
      <Text
        testID={`menu-section-item-text-${index}`}
        style={pressed ? styles.carouselItemSelected : styles.carouselItem}>
        {item.name.toUpperCase().slice(0, 11)}
      </Text>
    </TouchableOpacity>
  );
};

export default MenuSectionItem;
