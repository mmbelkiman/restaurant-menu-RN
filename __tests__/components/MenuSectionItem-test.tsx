import 'react-native';
import React from 'react';
import {fireEvent, render, RenderAPI} from '@testing-library/react-native';
import {IApiMenus} from '../../app/api/Menu';
import MenuSectionItem from '../../app/components/MenuSectionItem';

describe('MenuItem', () => {
  const item: IApiMenus = {
    name: 'My Menus',
    items: [{name: 'item 1', price: 123, description: 'my Desc', url: ''}],
  };
  const handleOnPress = jest.fn();

  const getRender = (): RenderAPI =>
    render(
      <MenuSectionItem
        item={item}
        index={0}
        pressed={false}
        onPress={handleOnPress}
      />,
    );

  it('Render Touchable', async () => {
    const element = await getRender().getByTestId(
      'menu-section-item-touchable-0',
    );
    expect(element).toBeTruthy();
    fireEvent(element, 'press');
    expect(handleOnPress).toBeCalledTimes(1);
  });

  it('Render Text', async () => {
    const element = await getRender().getByTestId('menu-section-item-text-0');
    expect(element).toBeTruthy();
    expect(element.children.join()).toEqual(item.name.toUpperCase());
  });

  it('Jest snapshot', async () => {
    const json = await getRender().toJSON();
    expect(json).toMatchSnapshot();
  });
});
