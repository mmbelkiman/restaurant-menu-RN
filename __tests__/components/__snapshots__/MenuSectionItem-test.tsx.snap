// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Jest snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginTop": 30,
      "opacity": 1,
    }
  }
  testID="menu-section-item-touchable-0"
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "borderBottomWidth": 0,
        "fontFamily": "Montserrat-Regular",
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
    testID="menu-section-item-text-0"
  >
    MY MENUS
  </Text>
</View>
`;
